geopandas/io/parquet.py

```diff
@@ -0,0 +1,223 @@
+        # retaining original column names
+        df = DataFrame(df.copy())
+
+        path, _, _, _ = get_filepath_or_buffer(path, mode="wb")
+
+        # Encode all geometry columns to WKB
+        for col in geometry_columns:
+            df[col] = df[col].apply(lambda g: g.wkb)
+
+        if index is None:
+            from_pandas_kwargs = {}
+        else:
+            from_pandas_kwargs = {"preserve_index": index}
+
+        table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
+
+        metadata = table.schema.metadata
+
+        # Store geopandas specific metadata
+        metadata.update(
+            {
+                "geo": json.dumps(
+                    {
+                        "crs": crs,
+                        "primary": geometry_column,
+                        "columns": geometry_columns,
+                    }
+                ).encode("utf-8")
+            }
+        )
+        table = table.replace_schema_metadata(metadata)
+
+        self.api.parquet.write_table(
+            table,
+            path,
+            compression=compression,
+            coerce_timestamps=coerce_timestamps,
+            **kwargs
+        )
+
+    def read(self, path, columns=None, **kwargs):
+        """Read from a parquet file into a GeoDataFrame, converting any geometry
+        columns present from WKB to GeoSeries.
+
+        See docstring for ``from_parquet``.
+        """
+
+        path, _, _, should_close = get_filepath_or_buffer(path)
+
+        kwargs["use_pandas_metadata"] = True
+        table = self.api.parquet.read_table(path, columns=columns, **kwargs)
+
+        df = table.to_pandas()
+
+        if should_close:
+            try:
+                path.close()
+            except:  # noqa: flake8
+                pass
+
+        metadata = table.schema.metadata
+
+        try:
+            geo_metadata = json.loads(metadata.get(b"geo", b"").decode("utf-8"))
+
+        except (TypeError, json.decoder.JSONDecodeError):
+            raise ValueError("Missing or malformed geo metadata in parquet file")
+
+        # Validate that required keys are present
+        required_keys = ("crs", "primary", "columns")
+        for key in required_keys:
+            if key not in geo_metadata:
+                raise ValueError("Geo metadata missing required key: {}".format(key))
+
+        # Convert the WKB columns that are present back to geometry
+        geometry_columns = df.columns.intersection(geo_metadata["columns"])
+
+        if columns and not len(geometry_columns):
+            warnings.warn(
+                """No geometry columns are included in the columns read from
+                the parquet file.  This will return a DataFrame instead of a
+                GeoDataFrame.""",
+                UserWarning,
+                stacklevel=2,
+            )
+
+            return df
+
+        for col in geometry_columns:
+            df[col] = GeoSeries(df[col].apply(lambda wkb: loads(wkb)))
```

Comment:
@jorisvandenbossche 

per your suggestion above (`geopandas.array.from_wkb`), this works fine if there is no index on the data frame read from the parquet file:
```
df[col] = GeoSeries(from_wkb(df[col].values))
```

However, it doesn't work if there is an index on the data frame:

```
>>> print(df)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/core/frame.py", line 655, in __repr__
    show_dimensions=show_dimensions,
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/core/frame.py", line 761, in to_string
    formatter.to_string()
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/io/formats/format.py", line 688, in to_string
    strcols = self._to_str_columns()
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/io/formats/format.py", line 626, in _to_str_columns
    fmt_values = self._format_col(i)
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/io/formats/format.py", line 817, in _format_col
    decimal=self.decimal,
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1033, in format_array
    return fmt_obj.get_result()
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1064, in get_result
    fmt_values = self._format_strings()
  File "/Users/bcward/.local/share/virtualenvs/geopandas-3VgNznKR/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1330, in _format_strings
    formatter = values._formatter(boxed=True)
  File "/Users/bcward/projects/geopandas/geopandas/array.py", line 1003, in _formatter
    xmin, ymin, xmax, ymax = self[~self.isna()][:10].total_bounds
  File "/Users/bcward/projects/geopandas/geopandas/array.py", line 752, in total_bounds
    b[:, 0].min(),  # minx
IndexError: too many indices for array
```

Specifically, the `GeoSeries` looks valid, but the data frame is not. 

Am I not using the `from_wkb` properly here?

Reply:
On the phone so can't check, but I think you can leave out the 'GeoSeries(', and directly assign the result of from_wkb to the column. 
Now you are running into the problem that the series that is created has not the same index as the df, so getting alignment issues.

(The error you get is an intetesting one, though, something I should investigate further)

Reply:
This solved the issue.  No need to wrap with `GeoSeries`.

Reply:
> (The error you get is an intetesting one, though, something I should investigate further)

Opened https://github.com/geopandas/geopandas/issues/1195 for this